common/src/main/scala/com/gu/zuora/encoding/CustomCodecs.scala

```diff
@@ -68,6 +67,13 @@ trait ModelsCodecs {
     .forProduct3("amount", "currency", "billingPeriod")(Contribution.apply)
     .or(Decoder.forProduct2("amount", "currency")((a: BigDecimal, c: Currency) => Contribution(a, c)))
   implicit val encodeContribution: Encoder[Contribution] = deriveEncoder
+
+  implicit val executionErrorCodec: Codec[ExecutionError] = deriveCodec
+  implicit val jsonWrapperDecoder: Decoder[JsonWrapper] = Decoder
+    .forProduct3("state", "error","encrypted")(JsonWrapper.apply)
```

Comment:
If encrypted was Option[Boolean] I don't think you'd need to complicate the decoder with the forProduct

Reply:
That's true, but it's not really an optional field, this is just a consequence of having to change 2 systems. I guess you could argue it either way but I thought it was better to hide the complexity in the decoder and keep it simple everywhere else.

Reply:
Fair enough.  Another option would be to remove the default value from support-models, and put it here.  Once this change has been rolled out we can follow up with a PR to remove the forPoduct2 making it a required field.

Reply:
Yes I think that probably would be better, it did occur to me that defining the default in support-models might not be the best place.

Reply:
Done